core/src/main/java/org/frankframework/parameters/Parameter.java

```diff
@@ -832,16 +836,21 @@ private Object getValueForFormatting(ParameterValueList alreadyResolvedParameter
 		Object substitutionValue = paramValue == null ? null : paramValue.getValue();
 
 		if (substitutionValue == null) {
-			Message substitutionValueMessage = session.getMessage(name);
-			if (!substitutionValueMessage.isEmpty()) {
-				if (substitutionValueMessage.asObject() instanceof Date) {
-					substitutionValue = preFormatDateType(substitutionValueMessage.asObject(), formatType, formatString);
+			Object substitutionValueMessage = session.get(name);
+			if (substitutionValueMessage != null) {
+				if (substitutionValueMessage instanceof Date) {
+					substitutionValue = preFormatDateType(substitutionValueMessage, formatType, formatString);
 				} else {
-					try {
-						substitutionValue = substitutionValueMessage.asString();
-					} catch (IOException e) {
-						throw new ParameterException(getName(), "Cannot get substitution value", e);
+					if (substitutionValueMessage instanceof String) {
+						substitutionValue = substitutionValueMessage;
+					} else {
+						try (Message message = Message.asMessage(substitutionValueMessage)) {
+							substitutionValue = message.asString();
+						} catch (IOException e) {
```

Comment:
Als het nu al een message is, sluit hij deze?

Reply:
Je hebt gelijk..dat is ook weer riskant. Ik draai dat weer terug en laat Message dan open (en hopelijk sluit iemand anders hem dan op de juiste plek).